src/lib/components/pages/settings/giftcards/settings.svelte

```diff
@@ -1,48 +1,97 @@
 <script lang="ts">
+	import { GRAPHQL_CLIENT } from '$lib/api/client';
+	import { operationStore } from '$lib/api/operation';
 	import {
+		GIFT_CARD_TAGS_QUERY,
+		GIFTCARD_EXPORT_MUTATION,
 		GIFTCARD_SETTINGS_QUERY,
 		GIFTCARD_SETTINGS_UPDATE_MUTATION,
+		GIFTCARD_TOTAL_COUNT_QUERY,
+		GIFT_CARD_BULK_CREATE_MUTATION,
 	} from '$lib/api/admin/giftcards';
-	import { GRAPHQL_CLIENT } from '$lib/api/client';
-	import { operationStore } from '$lib/api/operation';
+
 	import { SettingCog } from '$lib/components/icons';
+	import ShopCurrenciesSelect from '$lib/components/common/shop-currencies-select.svelte';
 	import { Alert } from '$lib/components/ui/Alert';
 	import { IconButton } from '$lib/components/ui/Button';
 	import { DropDown, type MenuItemProps } from '$lib/components/ui/Dropdown';
-	import { Input } from '$lib/components/ui/Input';
+	import { Checkbox, CheckboxSkeleton, Input, Label, RadioButton } from '$lib/components/ui/Input';
 	import { Modal } from '$lib/components/ui/Modal';
 	import { type DropdownTriggerInterface } from '$lib/components/ui/Popover';
-	import { Select, SelectSkeleton, type SelectOption } from '$lib/components/ui/select';
 	import {
+		GraphqlPaginableSelect,
+		Select,
+		SelectSkeleton,
+		type SelectOption,
+	} from '$lib/components/ui/select';
+
+	import {
+		ExportScope,
+		FileTypesEnum,
 		GiftCardSettingsExpiryTypeEnum,
 		TimePeriodTypeEnum,
+		type ExportGiftCardsInput,
 		type GiftCardSettingsUpdateInput,
 		type Mutation,
+		type MutationExportGiftCardsArgs,
 		type MutationGiftCardSettingsUpdateArgs,
 		type Query,
+		type QueryGiftCardsArgs,
+		type QueryGiftCardTagsArgs,
 	} from '$lib/gql/graphql';
+
 	import { CommonState } from '$lib/utils/common.svelte';
 	import { checkIfGraphqlResultHasError } from '$lib/utils/utils';
+	import { difference } from 'es-toolkit';
 	import { onMount } from 'svelte';
 
-	const GiftcardSettingsQuery = operationStore<Pick<Query, 'giftCardSettings'>>({
-		query: GIFTCARD_SETTINGS_QUERY,
-		requestPolicy: 'cache-and-network',
-		pause: true, // only execute when user choose to open setting modal
-	});
+	import GiftcardExpirationForm from './giftcard-expiration-form.svelte';
+	import { ALERT_MODAL_STORE } from '$lib/stores/ui/alert-modal';
+	import z, { array, number, object, string } from 'zod';
+
+	type Props = {
+		variables: QueryGiftCardsArgs;
+		selectedIds: Record<string, boolean>;
+	};
+	let { variables = $bindable(), selectedIds = $bindable() }: Props = $props();
 
 	let loading = $state(false);
+	let disabled = $state(false);
+
 	let openSettingModal = $state(false);
 	let openBulkIssueModal = $state(false);
 	let openExportCodesModal = $state(false);
+	let openExportBulkModal = $state(false);
+
+	let activeTags = $state<string[]>([]);
```

Comment:
GiftCardBulkCreateInput

the api has its own type.
Shy define individual field like this ???